### PR TITLE
NCTools documentation update for 2023.02

### PR DESCRIPTION
**Pull request by ngs333**
This PR updates the NCTools documentation by:

1) Slight improvements to the README.md file.
2) Updating the example "extreme frgrid" runscript to work on Gaea C5, and to run 5x faster that the
   prior one with similar amount of resources.
3) provides an outline one part of one fregrid remapping algorithm to highlight the use of some options (e.g. 
  cell_measures and cell_methods)
4) Provides a table/document of certain NCTools CI tests that are essentially showing a "workflow" and can provide the
   users with some context.
5) Adds a document concerning a trick for additional testing.

(NOTE:
Some files are being saved in the docs directory as pdf files. The source files for these are these google docs
which should have been shared with all:
https://docs.google.com/document/d/1bHB-_6HKofVTOT0xaX1MRZL_fUKKCxiuOL--zuDyymk
https://docs.google.com/document/d/17IgBJksVO297WAdMP_NRGh5i63BOADbUJ2U6HsE45NI/
https://docs.google.com/document/d/18a8_10_Y_3BGiPoVyYvbKkEUaj62KHsg9EBnWzQOMq0
)

### CODE REVIEW

**Comment by ngs333** [comment]
As per idea of @ceblanton , modified the README.md to mention fregrid_parallel, and also updated the
"extreme fregrid" documentation to clarify the purpose of fregrid_parallel,

File: README.md [hunk 1]
```diff
@@ -58,7 +58,11 @@ The tools available in FRE-NCtools are:
 
 
 ### Other Tools
-The [Ocean Model Grid Generator](https://github.com/NOAA-GFDL/ocean_model_grid_generator) can be copied or cloned from its GFDL homepage.
+There are several tools that have parallel versions and can overcome memory and cpu constrains of the serial conterpart.
+E.g. fregrid_parallel is used to generate remapping weights for high resolution grids (for further information, see the
```
Comment (bensonr) [inline README.md:62]: This implies fregrid_parallel can only create weight files.  I'd like to see the language clarified a bit.

File: docs/extreme_fregrid_sample_runscript.txt [hunk 2]
```diff
@@ -1,17 +1,19 @@
 ﻿# The NCTools app fregrid_parallel is the parallel version of fregrid, and
 # it is particularly useful for processing with large grids. fregrid will
-# generate this message when grids are too large for it :
+# generate this message when the remapping weights file is not available and
+# grids are too large for it to process:
```
Comment (bensonr) [inline docs/extreme_fregrid_sample_runscript.txt:4]: Is it possible to change the error message from fregrid to disambiguate what nxgrid really means.  We all know it refers to the number of exchange grid cells exceeding some hard-set limit.

**Review by bensonr — APPROVED** [review]

File: docs/extreme_fregrid_sample_runscript.txt (hunk 2 above)
Comment (ngs333) [reply docs/extreme_fregrid_sample_runscript.txt:4]: @bensonr 
I updated the message to this:

FATAL Error: The xgrid size is too large for resources.
 nxgrid is greater than MAXXGRID/nthreads; increase MAXXGRID,
 decrease nthreads, or increase number of MPI ranks.

The new related docs were also update, as your suggestions and those of @ceblanton.

**Review by bensonr — APPROVED** [review]

**Review by ceblanton — APPROVED** [review]

**Pull request merged**